File: misc/arithmetics/c/subs_no_subs/subs_no_subs.c
/**
 *
 *  Subtract without subtracting algorithm.
 *  Notice that there is not number control, so it's asumed that a is going to be
 *  bigger than b.
 *
 */

#include <stdio.h>

int	main(void)
{
  int a = 12;
  int b = 1;
  int baux = b;
  int result = 0;

  while (baux < a)
  {
	baux++;
	result++;
  }

  printf("Result of %d - %d = %d\n", a, b, result);
  return (0);
}

File: misc/arithmetics/javascript/subs_no_subs/subs_no_subs.js
/**
 *
 *  Subtract without subtracting algorithm.
 *  Notice that there is not number control, so it's asumed that a is going to be
 *  bigger than b.
 *
 */

const subs_no_subs = (a, b) => {

	let aux = b;
  	let i;

  	for(i = 0; aux < a; i++, aux++);

  console.log(`Result of ${a} - ${b} = ${i}`);
}

subs_no_subs(10, 3);
